src/screens/OtherProfile/elements/BioAndChat.js

```diff
@@ -0,0 +1,188 @@
+      await sentAnonDM();
+    } else {
+      await sendSignedDM();
+    }
+  };
+
+  return (
+    <View style={styles.bioAndSendChatContainer(isAnonimity)}>
+      <View style={styles.containerBio}>
+        {bio === null || bio === undefined ? (
+          <Text style={styles.bioText(isAnonimity)}>Send a message</Text>
+        ) : (
+          <Pressable onPress={openBio}>
+            <Text linkStyle={styles.seeMore} style={styles.bioText(isAnonimity)}>
+              {trimString(bio, 121)}{' '}
+              {bio.length > 121 ? <Text style={{color: colors.blue}}>see more</Text> : null}
+            </Text>
+          </Pressable>
+        )}
+      </View>
+      <TouchableOpacity
+        disabled={isSignedMessageEnabled}
+        activeOpacity={1}
+        onPress={showSignedMessageDisableToast}>
+        <TextAreaChat
+          isAnonimity={isAnonimity}
+          loadingAnonUser={loadingGenerateAnon}
+          avatarUrl={avatarUrl}
+          anonUser={anonProfile}
+          placeholder="Send a direct message"
+          disabledInput={!isSignedMessageEnabled}
+          onSend={onSendDM}
+          onChangeMessage={setDmChat}
+          disabledButton={loadingSendDM || !isSignedMessageEnabled || loadingGenerateAnon}
+          defaultValue={
+            isSignedMessageEnabled ? dmChat : `Only users ${username} follows can send messages`
+          }
+        />
+      </TouchableOpacity>
+      <TouchableOpacity onPress={toggleSwitch} style={styles.toggleSwitchContainer}>
+        <ToggleSwitch
+          value={isAnonimity}
+          onValueChange={toggleSwitch}
+          labelLeft={
+            isAnonimityEnabled || !isSignedMessageEnabled ? 'Anonymity' : 'Anonymity disabled'
+          }
+          styleLabelLeft={{color: isAnonimityEnabled ? colors.white : '#648ABF'}}
+        />
+      </TouchableOpacity>
+    </View>
+  );
+};
+
+const styles = StyleSheet.create({
+  bioAndSendChatContainer: (isAnonimity) => ({
+    backgroundColor: isAnonimity ? colors.bondi_blue : colors.blue1,
+    borderRadius: 15,
+    paddingHorizontal: 10,
+    paddingTop: 10
+  }),
+  containerBio: {
+    marginBottom: 10
+  },
+  bioText: (isAnonimity) => ({
+    color: isAnonimity ? colors.greenDark : colors.white,
+    fontSize: 14,
+    fontWeight: '600',
+    lineHeight: 22
+  }),
+  seeMore: {
+    fontFamily: fonts.inter[500],
+    fontSize: 14,
+    color: colors.black
+  },
+  toggleSwitchContainer: {
+    display: 'flex',
+    alignSelf: 'flex-end',
+    paddingVertical: 10
+  }
+});
+
+BioAndChat.propTypes = {
+  isAnonimity: PropTypes.bool,
+  bio: PropTypes.string,
+  openBio: PropTypes.bool,
+  dataMain: PropTypes.object,
+  isSignedMessageEnabled: PropTypes.bool,
+  showSignedMessageDisableToast: PropTypes.bool,
+  loadingGenerateAnon: PropTypes.bool,
+  avatarUrl: PropTypes.string,
+  anonProfile: PropTypes.object,
+  username: PropTypes.string,
+  toggleSwitch: PropTypes.func,
+  isAnonimityEnabled: PropTypes.bool
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The PropTypes validation for the props is good but some of them are marked as optional while they seem to be required for the component to work correctly. For example, `dataMain`, `anonProfile`, `username`, and `toggleSwitch` should probably be marked as required.
```diff
- dataMain: PropTypes.object,
+ dataMain: PropTypes.object.isRequired,
- anonProfile: PropTypes.object,
+ anonProfile: PropTypes.object.isRequired,
- username: PropTypes.string,
+ username: PropTypes.string.isRequired,
- toggleSwitch: PropTypes.func,
+ toggleSwitch: PropTypes.func.isRequired,
```

Overall, the component is well structured and follows good practices. The state and effects are managed correctly, and the component is split into smaller functions which improves readability and maintainability.


<!-- This is an auto-generated comment by OSS CodeRabbit -->

Reply:
resolved in new commit